flytekit/core/python_function_task.py

```diff
@@ -63,12 +64,63 @@ def __init__(
+          executed in serial when caching is enabled. This means that given multiple concurrent executions over
+          identical inputs, only a single instance executes and the rest wait to reuse the cached results. This
+          parameter does nothing without also setting the cache parameter.
+        :param cache_version: Cache version to use. Changes to the task signature will automatically trigger a cache miss,
+           but you can always manually update this field as well to force a cache miss. You should also manually bump
+           this version if the function body/business logic has changed, but the signature hasn't.
+        :param retries: Number of times to retry this task during a workflow execution.
+        :param interruptible: [Optional] Boolean that indicates that this task can be interrupted and/or scheduled on nodes
+                              with lower QoS guarantees. This will directly reduce the `$`/`execution cost` associated,
+                              at the cost of performance penalties due to potential interruptions. Requires additional
+                              Flyte platform level configuration. If no value is provided, the task will inherit this
+                              attribute from its workflow, as follows:
+                              No values set for interruptible at the task or workflow level - task is not interruptible
+                              Task has interruptible=True, but workflow has no value set - task is interruptible
+                              Workflow has interruptible=True, but task has no value set - task is interruptible
+                              Workflow has interruptible=False, but task has interruptible=True - task is interruptible
+                              Workflow has interruptible=True, but task has interruptible=False - task is not interruptible
+        :param deprecated: A string that can be used to provide a warning message for deprecated task. Absence / empty str
+                           indicates that the task is active and not deprecated
+        :param timeout: the max amount of time for which one execution of this task should be executed for. The execution
+                        will be terminated if the runtime exceeds the given timeout (approximately).
         """
-        super().__init__(name=name, task_config=task_config, task_type=task_type, task_resolver=task_resolver, **kwargs)
+        _metadata = TaskMetadata(
+            cache=cache,
+            cache_serialize=cache_serialize,
+            cache_version=cache_version,
+            retries=retries,
+            interruptible=interruptible,
+            deprecated=deprecated,
+            timeout=timeout,
+        )
+        super().__init__(
+            name=name,
+            task_config=task_config,
+            task_type=task_type,
+            task_resolver=task_resolver,
+            metadata=_metadata,
+            **kwargs,
```

Comment:
we should ensure that metadata does not exist in kwargs, as that would break things. 
the thing is many folks use metadata directly! 
Another problem with this is that if we change the metadata it will break as we will have to keep adding new things.

Reply:
People are already using metadata. This will fail weirdly.

My recommendation add these fields to InstanceTask and then check for either these or metadata.
Also `metadata` field has higher precedence